infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/convert/DockerImageEnvironmentConverter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.environment.convert;
+
+import static java.lang.String.format;
+import static org.eclipse.che.workspace.infrastructure.openshift.Constants.MACHINE_NAME_ANNOTATION_FMT;
+
+import com.google.common.collect.ImmutableMap;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalMachineConfig;
+import org.eclipse.che.workspace.infrastructure.docker.environment.dockerimage.DockerImageEnvironment;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenShiftEnvironment;
+
+/**
+ * Converts {@link DockerImageEnvironment} to {@link OpenShiftEnvironment}.
+ *
+ * @author Sergii Leshchenko
+ * @author Anton Korneta
+ */
+@Singleton
+public class DockerImageEnvironmentConverter {
+
+  static final String POD_NAME = "dockerimage";
+  static final String CONTAINER_NAME = "container";
+
+  public OpenShiftEnvironment convert(DockerImageEnvironment environment)
+      throws ValidationException {
+    final Map<String, InternalMachineConfig> machines = environment.getMachines();
+    if (!machines.keySet().iterator().hasNext()) {
+      throw new ValidationException("Environment must contain at least one machine configuration");
```

Comment:
As for me InternalInfrastructureException is better fit here since it is really abnormal situation